src/api/cow/cow.spec.ts

```diff
@@ -0,0 +1,441 @@
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    })
+  }
+})
+
+test('Valid: Get last 5 trades for a given trader ', async () => {
+  const TRADES_RESPONSE = Array(5).fill(TRADE_RESPONSE)
+  fetchMock.mockResponseOnce(JSON.stringify(TRADES_RESPONSE), { status: HTTP_STATUS_OK })
+  const trades = await cowSdk.cowApi.getTrades({
+    owner: '0x00000000005ef87f8ca7014309ece7260bbcdaeb', // Trader
+    limit: 5,
+    offset: 0,
+  })
+  expect(fetchMock).toHaveBeenCalledTimes(1)
+  expect(fetchMock).toHaveBeenCalledWith(
+    'https://api.cow.fi/rinkeby/api/v1/trades?owner=0x00000000005ef87f8ca7014309ece7260bbcdaeb&limit=5',
+    {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    }
+  )
+  expect(trades.length).toEqual(5)
+})
+
+test('Invalid: Get last 5 trades for an unexisting trader ', async () => {
+  fetchMock.mockResponseOnce(
+    JSON.stringify({
+      errorType: 'NotFound',
+      description: "You've passed an invalid URL",
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+
+  try {
+    await cowSdk.cowApi.getTrades({
+      owner: 'invalidOwner',
+      limit: 5,
+      offset: 0,
+    })
+  } catch (e: unknown) {
+    const error = e as OperatorError
+    expect(error.message).toEqual('Token pair selected has insufficient liquidity')
+    expect(fetchMock).toHaveBeenCalledTimes(1)
+    expect(fetchMock).toHaveBeenCalledWith('https://api.cow.fi/rinkeby/api/v1/trades?owner=invalidOwner&limit=5', {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    })
+  }
+})
+
+test('Valid: Get Price Quote (Legacy)', async () => {
+  fetchMock.mockResponseOnce(JSON.stringify(PRICE_QUOTE_RESPONSE), { status: HTTP_STATUS_OK })
+  const price = await cowSdk.cowApi.getPriceQuoteLegacy({
+    baseToken: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+    quoteToken: '0x6810e776880c02933d47db1b9fc05908e5386b96',
+    amount: '1234567890',
+    kind: OrderKind.BUY,
+  } as PriceQuoteParams)
+  expect(fetchMock).toHaveBeenCalledTimes(1)
+  expect(fetchMock).toHaveBeenCalledWith(
+    'https://api.cow.fi/rinkeby/api/v1/markets/0x6810e776880c02933d47db1b9fc05908e5386b96-0x6810e776880c02933d47db1b9fc05908e5386b96/buy/1234567890',
+    {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    }
+  )
+  expect(price?.amount).toEqual('1234567890')
+  expect(price?.token).toEqual('0x6810e776880c02933d47db1b9fc05908e5386b96')
```

Comment:
I also suggest to use the mock data object rather than hardcoding to make it easier in case we need to update those values